ROX-26041: Add required Date header to email messages

vjwilson opened the pull request:
### Description

This changes adds the required Date header to all emails send by the ACS system.

(from the Jira ticket description)
> In ACS the emails that are sent from the "Vulnerability Management (2.0)" > Vulnerability Reporting function are missing the Date header in the email.
> 
>This header is mandatory in emails as specified by the RFC 5322
> 
> https://datatracker.ietf.org/doc/html/rfc5322#section-3.6
> 
> Mails without a date header are still processed but gather a higher than normal spam score due to RFC non-compliance. Additionally the mails are not properly ordered in the email client (eg. Thunderbird) because of the missing date. This can create confusion on the time a vulnerability report has actually been generated.

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request)  is not needed

### Testing and quality


- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [x] modified existing tests

#### How I validated my change

* Added additional checks for this new header to this function's unit tests
* Verified CI
* Manually, by:
* * Deploying the PR build image
* * Creating a email integration, a collection, and a VM report that uses those
* * Selecting "Send now" for that VM report
* * Inspecting the raw headers on the email I received (I saved the email as a .eml file, which I opened in a text editor.)

Relevant section of email headers:
```
From: ACS <scooby@vanwilson.info>
To: vwilson@redhat.com
Date: 2024-10-01T23:57:11Z
Subject: StackRox Workload CVE Report for Van3; Scope: Scooby
MIME-Version: 1.0
```

Note: I had opened the headers of a previous test email, without this change, and there was no date header.

clickboo reviewed (approved):
Assuming this will be so for compliance report emails as well

vjwilson commented:
@clickboo ,

> Assuming this will be so for compliance report emails as well

Yes, I can confirm this because when I modified the Subject header for a previous bug, I modified the same function in this same `email.go` file, and tested that it worked for both VM Reports and Compliance Report emails.

vjwilson commented:
/test gke-qa-e2e-tests
